jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/main/java/org/jbpm/workbench/pr/backend/server/util/VariableHelper.java

```diff
@@ -80,7 +80,16 @@ public static void registerProcessor(VariableProcessor processor) {
 
         if (!filteredVariables.isEmpty()) {
             filteredVariables.forEach(variable -> {
-                variablesSummary.add(new ProcessVariableSummary(variable.getVariableName(), variable.getVariableName(), variable.getProcessInstanceId(), variable.getOldValue(), variable.getValue(), variable.getDate().getTime(), ""));
+                ProcessVariableSummary variableSummary = new ProcessVariableSummary(variable.getVariableName(),
```

Comment:
I would suggest using the constructor to pass all variables and also transform this into a stream style, using filteredVariables.map. The map function can be an isolated function to be tested independently.

Reply:
oks